src/traits.rs

```diff
@@ -220,4 +253,44 @@ pub mod test_helpers {
             }
         }
     }
+
+    #[allow(non_snake_case)]
+    /// Check that bad polynomial lengths are properly caught as errors
+    pub fn bad_polynomial_length<Signer: super::Signer, Aggregator: super::Aggregator>() {
+        let Nk: u32 = 10;
+        let Np: u32 = 4;
+        let T: u32 = 7;
+        let signer_ids: Vec<Vec<u32>> = vec![vec![1, 2, 3], vec![4, 5], vec![6, 7, 8, 9], vec![10]];
+        let mut rng = OsRng;
+        let mut signers: Vec<Signer> = signer_ids
+            .iter()
+            .enumerate()
+            .map(|(id, ids)| {
+                if *ids == vec![10] {
+                    Signer::new(id.try_into().unwrap(), ids, Nk, Np, T - 1, &mut rng)
```

Comment:
Thanks for adding this, but could you also add a test for the case where a malicious signer creates a polynomial whose degree is greater than the threshold?

Reply:
Done @technovision99